keg_storage/backends/base.py

```diff
@@ -10,18 +63,33 @@ class StorageBackend:
     def __init__(self, *args, **kwargs):
         pass
 
-    def list(self, path):
-        """Returns an iterator over the given path"""
+    def list(self, path: str) -> typing.List[ListEntry]:
+        """
+        Returns a list of string keys available under the directory or prefix given
+        """
         raise NotImplementedError()
 
-    def get(self, path, dest):
+    def open(self, path: str, mode: FileMode) -> RemoteFile:
         raise NotImplementedError()
 
-    def put(self, path, dest):
+    def delete(self, path: str):
         raise NotImplementedError()
 
-    def delete(self, path):
-        raise NotImplementedError()
+    def get(self, path, dest):
+        with self.open(path, FileMode.read) as infile:
+            with open(dest, 'wb') as outfile:
+                for chunk in infile.iter_chunks():
+                    outfile.write(chunk)
+
+    def put(self, path, dest):
+        buffer_size = 5 * 1024 * 1024
+
+        with self.open(dest, FileMode.write) as outfile:
+            with open(path, 'rb') as infile:
+                buf = infile.read(buffer_size)
```

Comment:
Because `infile` is a normal local python file which will be iterated line by line. We know nothing about the contents so it would be a bad idea to attempt to read a full line. What is possible is `for chunk in iter(lambda: infile.read(buffer_size), b''):` but I don't know if that does anything to improve clarity.